File: gin-vue-admin-main/server/router/wl_playform/wl_resources.go
package wl_playform

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type WlResourcesRouter struct{}

// InitWlResourcesRouter 初始化 wlResources 路由信息
func (s *WlResourcesRouter) InitWlResourcesRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	wlResourcesRouter := Router.Group("wlResources").Use(middleware.OperationRecord())
	wlResourcesRouterWithoutRecord := Router.Group("wlResources")
	// wlResourcesRouterWithoutAuth := PublicRouter.Group("wlResources")

	var wlResourcesApi = v1.ApiGroupApp.Wl_playformApiGroup.WlResourcesApi
	{
		wlResourcesRouter.POST("createWlResources", wlResourcesApi.CreateWlResources)                    // 新建wlResources
		wlResourcesRouter.POST("createWithTransaction", wlResourcesApi.CreateWlResourcesWithTransaction) // 新建资源（事务处理）
		wlResourcesRouter.DELETE("deleteWlResources", wlResourcesApi.DeleteWlResources)                  // 删除wlResources
		wlResourcesRouter.DELETE("deleteWlResourcesByIds", wlResourcesApi.DeleteWlResourcesByIds)        // 批量删除wlResources
		wlResourcesRouter.PUT("updateWlResources", wlResourcesApi.UpdateWlResources)                     // 更新wlResources
		wlResourcesRouter.POST("verifyWlResources", wlResourcesApi.VerifyWlResources)                    // 验证wlResources
	}
	{
		wlResourcesRouterWithoutRecord.GET("findWlResources", wlResourcesApi.FindWlResources)       // 根据ID获取wlResources
		wlResourcesRouterWithoutRecord.GET("getWlResourcesList", wlResourcesApi.GetWlResourcesList) // 获取wlResources列表
	}
	// {
	// 	wlResourcesRouterWithoutAuth.GET("getWlResourcesPublic", wlResourcesApi.GetWlResourcesPublic) // 获取wlResources列表
	// }
}

File: gin-vue-admin-main/web/src/api/wl_playform/wlResources.js
import service from '@/utils/request'

// @Tags WlResources
// @Summary 创建资源（事务处理）
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.CreateWlResourcesRequest true "创建资源"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /wlResources/createWithTransaction [post]
export const createWlResourcesWithTransaction = (data) => {
  return service({
    url: '/wlResources/createWithTransaction',
    method: 'post',
    data
  })
}

// @Tags WlResources
// @Summary 创建资源
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body wl_playform.WlResources true "创建资源"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /wlResources/createWlResources [post]
export const createWlResources = (data) => {
  return service({
    url: '/wlResources/createWlResources',
    method: 'post',
    data
  })
}

// @Tags WlResources
// @Summary 删除资源
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body wl_playform.WlResources true "删除资源"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /wlResources/deleteWlResources [delete]
export const deleteWlResources = (params) => {
  return service({
    url: '/wlResources/deleteWlResources',
    method: 'delete',
    params
  })
}

// @Tags WlResources
// @Summary 批量删除资源
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /wlResources/deleteWlResources [delete]
export const deleteWlResourcesByIds = (params) => {
  return service({
    url: '/wlResources/deleteWlResourcesByIds',
    method: 'delete',
    params
  })
}

// @Tags WlResources
// @Summary 更新资源
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body wl_playform.WlResources true "更新资源"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /wlResources/updateWlResources [put]
export const updateWlResources = (data) => {
  return service({
    url: '/wlResources/updateWlResources',
    method: 'put',
    data
  })
}

// @Tags WlResources
// @Summary 用id查询资源
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query wl_playform.WlResources true "用id查询资源"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /wlResources/findWlResources [get]
export const findWlResources = (params) => {
  return service({
    url: '/wlResources/findWlResources',
    method: 'get',
    params
  })
}

// @Tags WlResources
// @Summary 分页获取资源列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.WlResourcesSearch true "分页获取资源列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /wlResources/getWlResourcesList [get]
export const getWlResourcesList = (params) => {
  return service({
    url: '/wlResources/getWlResourcesList',
    method: 'get',
    params
  })
}

// @Tags WlResources
// @Summary 验证资源
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body wl_playform.WlResources true "验证资源"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"验证成功"}"
// @Router /wlResources/verifyWlResources [post]
export const verifyWlResources = (data) => {
  return service({
    url: '/wlResources/verifyWlResources',
    method: 'post',
    data
  })
}

// @Tags WlResources
// @Summary 不需要鉴权的wlResources表接口
// @Accept application/json
// @Produce application/json
// @Param data query wl_playformReq.WlResourcesSearch true "分页获取wlResources表列表"
// @Success 200 {object} response.Response{data=object,msg=string} "获取成功"
// @Router /wlResources/getWlResourcesPublic [get]
export const getWlResourcesPublic = () => {
  return service({
    url: '/wlResources/getWlResourcesPublic',
    method: 'get',
  })
}
